src/github/github.api.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { gql, GraphQLClient } from 'graphql-request';
+import { User } from './interfaces/user.interface';
+import { GetContributorInfoQuery } from './types/queries';
+
+@Injectable()
+export class GithubApi {
+  constructor(private readonly configService: ConfigService) {}
```

Comment:
```suggestion
  constructor(private readonly config: ConfigService) {}
```

Avoid suffixing the provider with its type when there is no risk of confusion (e.g. "configService" => "config", "githubService" => "github" ...)

The inverse is also true: say you are in a file that uses a "githubRepository", a "githubApi" and a "githubDao", the naming should be "repository", "api", and "dao" because it is clear that the scope is github